examples/research_projects/jax-projects/wav2vec2/run_wav2vec2_pretrain_flax.py

```diff
@@ -329,14 +328,20 @@ def main():
         model_args.model_name_or_path, cache_dir=model_args.cache_dir, do_normalize=True
     )
 
+    # check that all files have the correct sampling rate
+    datasets = datasets.cast_column(data_args.audio_file_column, Audio(sampling_rate=feature_extractor.sampling_rate))
+
     def prepare_dataset(batch):
-        # check that all files have the correct sampling rate
-        batch["speech"], _ = librosa.load(batch[data_args.speech_file_column], sr=feature_extractor.sampling_rate)
+        batch["speech"] = batch[data_args.audio_file_column]["array"]
         return batch
 
+    remove_columns_values = datasets["train"].column_names.copy()
+    if "speech" in set(remove_columns_values):
+        remove_columns_values.remove("speech")
+
     # load audio files into numpy arrays
     vectorized_datasets = datasets.map(
-        prepare_dataset, num_proc=data_args.preprocessing_num_workers, remove_columns=datasets["train"].column_names
+        prepare_dataset, num_proc=data_args.preprocessing_num_workers, remove_columns=remove_columns_values
```

Comment:
Why change this? It was fine before no?

Reply:
in function preprocess, the column will be assigned to colum name `speech`, I think maybe the bug can occur seen some data have the column `speech` will automatic remove, this is not a rare case seen something I name my audio column name as `speech`